pkg/ifuzz/ifuzz_test.go

```diff
@@ -116,6 +116,46 @@ func testDecode(t *testing.T, arch string) {
 	}
 }
 
+func TestGenerate(t *testing.T) {
+	for _, arch := range allArches {
+		t.Run(arch, func(t *testing.T) {
+			testGenerate(t, arch)
+		})
+	}
+}
+
+func testGenerate(t *testing.T, arch string) {
+	insnset := iset.Arches[arch]
+	r := rand.New(testutil.RandSource(t))
+	for mode := iset.Mode(0); mode < iset.ModeLast; mode++ {
+		for repeat := 1; repeat < 10; repeat++ {
+			if len(insnset.GetInsns(mode, iset.TypeUser)) == 0 {
```

Comment:
What arch does not have TypeUser instructions? Is it powerpc?

Reply:
If generation for this arch crashes, we cannot use it. So what's happening here?

Reply:
This is because not all architectures support all modes. x86 supports `ModeLong64`, `ModeProt32`, `ModeProt16`, and `ModeReal16`, whereas powerpc only supports `ModeLong64` and `ModeProt32`. For arm64 I was initially planning to only provide `ModeLong64` support.

Overall, a better solution might be to let every arch list the supported modes and only loop over them, not everything between 0 and ModeLast.

Reply:
Ah, I see, I missed this modes/arches relation.
Let's leave it as is for now. Nothing is breaking and we are not making thing worse.